### ISSUE TITLE
Possible Arithmetic Overflow in ListBox.h

### ISSUE THREAD

**Issue by Brett208** [OutpostUniverse/OPHD#1031]
I am trying to resolve an MSVC warning concerning a possible arithmetic overflow, but I am having trouble figuring out what the problem is. The 2 variables being highlighted appear to both be of `int` type?

I think it is highlighting `y` and `mScrollArea.y`, which should both int values. `mScrollArea` is a `NAS2D::Rectangle<int>`. Inside `Rectangle`, `y` is defined as `BaseType` which should be set to int in this case. 

See screen shot below from IDE:

<img width="724" alt="PossibleArithmeticOverflow" src="https://user-images.githubusercontent.com/15183337/124400067-d2631200-dced-11eb-8d0f-d8d1ae55ef67.png">

Related to #320

**Comment by Brett208** [comment]
Ah, I think I just figured out what MSVC was pointing out. Since the IDE knows the value is eventually being cast into an integer with larger storage, it wants the type cast to occur before the operation is performed. I think this is because it is trying to reduce the chance of an integer overflow occurring.

**Comment by DanRStevens** [comment]
Sounds like you figured it out. I'm curious though, this warning doesn't seem to be present on the CI builds. There there some additional warning flag we should be turning on?

**Comment by Brett208** [comment]
This one was generated by code analysis as discussed in #320. A very brief google search didn't seem to illuminate how to make code analysis to run on AppVeyor. They tend to show up in the IDE when you have the file open and compile. A full list doesn't populate unless you manually run the command. I'm not sure what the exact threshold is to trigger them showing up in the IDE with running the full analysis.

I am trying to clear them when they show up to keep the compiler warning list low. Otherwise I tend to miss sometimes more relevant warnings that I may be causing during development.

All the rest showing up without manually compiling are related to the non-class enums in the code base.

I think there are ways to further customize the code analysis and by default it doesn't run in the most strict settings. It reminds me more of static code analysis than just strictly compiler warnings.

**Comment by DanRStevens** [comment]
For some reason I thought it was already enabled. I thought some of the warning we were getting were from the Code Analysis option.

I noticed in the project file, there is:
```
<CodeAnalysisRuleSet>NativeRecommendedRules.ruleset</CodeAnalysisRuleSet>
```

Curiously though, that's only set for the Debug builds. We only build Release on CI due to only having a single concurrent build and long build times.

Anyway, I found this, which may be related:
https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-configure-code-analysis-for-a-managed-code-project?view=vs-2019
